app/src/main/java/com/eveningoutpost/dexdrip/G5Model/Ob1G5StateMachine.java

```diff
@@ -1218,8 +1218,7 @@ private static void processBacksies(List<BackFillStream.Backsie> backsies) {
                 UserError.Log.wtf(TAG, "Backfill timestamp unrealistic: " + JoH.dateTimeText(time) + " (ignored)");
             } else {
                 if (BgReading.getForPreciseTimestamp(time, Constants.MINUTE_IN_MS * 4) == null) {
-                    final BgReading bgr = BgReading.bgReadingInsertFromG5(backsie.getGlucose(), time);
-                    if (bgr != null) bgr.appendSourceInfo("Backfill");
+                    final BgReading bgr = BgReading.bgReadingInsertFromG5(backsie.getGlucose(), time, "Backfill");
```

Comment:
Why change the method signature when can just use the `bgr.appendSourceInfo("Backfill");` ?

Reply:
From what I could tell, using appendSourceInfo after the insertfromg5 method didn't save the entry in the database, only modified the local instance of bgreading. So by doing this it runs appendSourceInfo before the .save(), and I don't have to explicitly call save after which seems like something internal to the method that external accessors shouldn't have to worry about.

Reply:
I think you're right about that and its effectively a bug as the appended source info *is* intended to be persistent. Other than doing multiple saves, your refactoring to put it in the constructor seems to be a good way of doing it given that `bgReadingInsertFromG5` is only called in two places.